### PR TITLE
Convert the abstract class DAOs to interfaces.

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

- Convert the Java abstract class DAOs to interfaces. 

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vkay94** [comment]
Just of curiosity:  I know that since NP uses Java 8 now and therefore because of the `default` feature the conversion is possible now (without side-effects I think). Are there benefits of switching from abstract classes to interfaces (except of getting rid of `public abstract`)?

**Comment by Isira-Seneviratne** [comment]
> Just of curiosity: I know that since NP uses Java 8 now and therefore because of the `default` feature the conversion is possible now (without side-effects I think). Are there benefits of switching from abstract classes to interfaces (except of getting rid of `public abstract`)?

Aside from the change you mentioned, no.

**Review by mhmdanas — APPROVED** [review]

File: app/src/main/java/org/schabi/newpipe/database/stream/dao/StreamStateDAO.java [hunk 1]
```diff
@@ -17,31 +17,31 @@
...
-    public abstract Flowable<List<StreamStateEntity>> getState(long streamId);
+    Flowable<List<StreamStateEntity>> getState(long streamId);
 
     @Query("DELETE FROM " + STREAM_STATE_TABLE + " WHERE " + JOIN_STREAM_ID + " = :streamId")
-    public abstract int deleteState(long streamId);
+    int deleteState(long streamId);
 
     @Insert(onConflict = OnConflictStrategy.IGNORE)
-    abstract void silentInsertInternal(StreamStateEntity streamState);
+    void silentInsertInternal(StreamStateEntity streamState);
```
Comment (mhmdanas) [inline app/src/main/java/org/schabi/newpipe/database/stream/dao/StreamStateDAO.java:41]: The only issue I see with this change is that "internal" methods like this one have to be public (all of an interface's members can be _only_ public).

Comment (Isira-Seneviratne) [reply app/src/main/java/org/schabi/newpipe/database/stream/dao/StreamStateDAO.java:41]: Private interface methods were introduced in Java 9.

Comment (mhmdanas) [reply app/src/main/java/org/schabi/newpipe/database/stream/dao/StreamStateDAO.java:41]: @Isira-Seneviratne indeed, you're right; I was not aware of that. However, we're using Java 8 here, so we still can't do that.

**Comment by TobiGr** [comment]
I think it is a reasonable change (in other word: yes)

**Review by Stypox — APPROVED** [review]

**Pull request merged**